### PR TITLE
Adding rabbitmq setup

### PR DESCRIPTION
**Pull request by Nexengineer**
Please dont merge it now. I will merge it

### CODE REVIEW

**Comment by ganesanarun** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 3
           

Complexity increasing per file
==============================
- rabbitmq/rabbitmqsetup.py  1
         

Complexity decreasing per file
==============================
+ src/main/java/in/projecteka/consentmanager/consent/HiuConsentNotificationListener.java  -2
+ src/main/java/in/projecteka/consentmanager/dataflow/PostDataFlowRequestApproval.java  -2
+ src/main/java/in/projecteka/consentmanager/consent/ConsentNotificationPublisher.java  -2
+ src/main/java/in/projecteka/consentmanager/consent/PostConsentRequest.java  -2
+ src/main/java/in/projecteka/consentmanager/consent/ConsentRequestNotificationListener.java  -2
+ src/main/java/in/projecteka/consentmanager/consent/HipConsentNotificationListener.java  -2
+ src/main/java/in/projecteka/consentmanager/dataflow/DataFlowBroadcastListener.java  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/ProjectEKA/consent-manager/pullRequest?prid=5556146&bid=18221425)

**Comment by ganesanarun** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 3
           

Complexity increasing per file
==============================
- rabbitmq/rabbitmqsetup.py  1
         

Complexity decreasing per file
==============================
+ src/main/java/in/projecteka/consentmanager/consent/HipConsentNotificationListener.java  -2
+ src/main/java/in/projecteka/consentmanager/dataflow/PostDataFlowRequestApproval.java  -2
+ src/main/java/in/projecteka/consentmanager/consent/ConsentRequestNotificationListener.java  -2
+ src/main/java/in/projecteka/consentmanager/consent/ConsentNotificationPublisher.java  -2
+ src/main/java/in/projecteka/consentmanager/dataflow/DataFlowBroadcastListener.java  -2
+ src/main/java/in/projecteka/consentmanager/consent/HiuConsentNotificationListener.java  -2
+ src/main/java/in/projecteka/consentmanager/consent/PostConsentRequest.java  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/ProjectEKA/consent-manager/pullRequest?prid=5556146&bid=18221425)

**Comment by ganesanarun** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 3
           

Complexity increasing per file
==============================
- rabbitmq/rabbitmqsetup.py  1
         

Complexity decreasing per file
==============================
+ src/main/java/in/projecteka/consentmanager/consent/ConsentNotificationPublisher.java  -2
+ src/main/java/in/projecteka/consentmanager/consent/PostConsentRequest.java  -2
+ src/main/java/in/projecteka/consentmanager/dataflow/DataFlowBroadcastListener.java  -2
+ src/main/java/in/projecteka/consentmanager/dataflow/PostDataFlowRequestApproval.java  -2
+ src/main/java/in/projecteka/consentmanager/consent/HiuConsentNotificationListener.java  -2
+ src/main/java/in/projecteka/consentmanager/consent/ConsentRequestNotificationListener.java  -2
+ src/main/java/in/projecteka/consentmanager/consent/HipConsentNotificationListener.java  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/ProjectEKA/consent-manager/pullRequest?prid=5556146&bid=18221425)

File: rabbitmq/definitions.json [hunk 1]
```diff
@@ -0,0 +1 @@
...: "topic", "auto_delete": false}], "bindings": [{"vhost": "/", "destination": "cm-dead-letter-queue", "routing_key": "#", "source": "cm-dead-letter-exchange", "arguments": {}, "destination_type": "queue"}, {"vhost": "/", "destination": "hip-data-flow-request-queue", "routing_key": "hip-data-flow-request-queue", "source": "exchange", "arguments": {}, "destination_type": "queue"}, {"vhost": "/", "destination": "hip-consent-notification-queue", "routing_key": "hip-consent-notification-queue", "source": "exchange", "arguments": {}, "destination_type": "queue"}, {"vhost": "/", "destination": "hiu-consent-notification-queue", "routing_key": "hiu-consent-notification-queue", "source": "exchange", "arguments": {}, "destination_type": "queue"}, {"vhost": "/", "destination": "consent-request-queue", "routing_key": "consent-request-queue", "source": "exchange", "arguments": {}, "destination_type": "queue"}], "rabbit_version": "3.8.3", "permissions": [{"write": ".*", "vhost": "/", "read": ".*", "user": "guest", "configure": ".*"}], "global_parameters": [{"name": "cluster_name", "value": "rabbit@rabbitmq"}, {"name": "internal_cluster_id", "value": "rabbitmq-cluster-id-TmPfev8PqMAODZPsBGdQSg"}]}
```
Comment (ganesanarun) [inline rabbitmq/definitions.json:1]: what is this?

File: docker-compose-rabbitmq.yml [hunk 2]
```diff
@@ -0,0 +1,20 @@
+version: '3'
+services:
+  rabbitmq:
+    #      build: ./rabbitmq/configuration
+    image: rabbitmq:3.8.3-management
+    container_name: rabbitmq_local
+    restart: always
+    hostname: "rabbitmq_local"
```
Comment (ganesanarun) [inline docker-compose-rabbitmq.yml:8]: if we use this container\_name/host\_name(rabbitmq\_local), would not change the connection string in the code?

File: rabbitmq/definitions.json (hunk 1 above)
Comment (Nexengineer) [reply rabbitmq/definitions.json:1]: This user config, I have used the same which I exported from rabbitmq in the production environment(aws)

File: docker-compose-rabbitmq.yml (hunk 2 above)
Comment (Nexengineer) [reply docker-compose-rabbitmq.yml:8]: This will be changed as per code

**Comment by ganesanarun** [comment]
Can you add more info about how does it work in the README?

Let's say, I as a developer need to have new queue, what should I do? How it will get added in the higher environment even if there's a command I have to run manually by getting into box meantime?

**Comment by ganesanarun** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 3
           

Complexity increasing per file
==============================
- src/main/java/in/projecteka/consentmanager/dataflow/DataFlowBroadcastListener.java  1
- src/main/java/in/projecteka/consentmanager/dataflow/PostDataFlowRequestApproval.java  1
- src/main/java/in/projecteka/consentmanager/consent/ConsentNotificationPublisher.java  1
- src/main/java/in/projecteka/consentmanager/consent/HiuConsentNotificationListener.java  1
- rabbitmq/rabbitmqsetup.py  1
- src/main/java/in/projecteka/consentmanager/consent/HipConsentNotificationListener.java  1
- src/main/java/in/projecteka/consentmanager/consent/ConsentRequestNotificationListener.java  1
- src/main/java/in/projecteka/consentmanager/consent/PostConsentRequest.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/ProjectEKA/consent-manager/pullRequest?prid=5556146&bid=18221425)

**Comment by ganesanarun** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/main/java/in/projecteka/consentmanager/dataflow/DataFlowBroadcastListener.java  1
- src/main/java/in/projecteka/consentmanager/consent/HiuConsentNotificationListener.java  1
- src/main/java/in/projecteka/consentmanager/consent/ConsentNotificationPublisher.java  1
- src/main/java/in/projecteka/consentmanager/dataflow/PostDataFlowRequestApproval.java  1
- src/main/java/in/projecteka/consentmanager/ConsentManagerConfiguration.java  1
- src/main/java/in/projecteka/consentmanager/consent/HipConsentNotificationListener.java  1
- src/main/java/in/projecteka/consentmanager/consent/PostConsentRequest.java  1
- src/main/java/in/projecteka/consentmanager/consent/ConsentRequestNotificationListener.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/ProjectEKA/consent-manager/pullRequest?prid=5556146&bid=18221425)

**Comment by ganesanarun** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity decreasing per file
==============================
+ src/main/java/in/projecteka/consentmanager/consent/ConsentNotificationPublisher.java  -2
+ src/main/java/in/projecteka/consentmanager/dataflow/PostDataFlowRequestApproval.java  -2
+ src/main/java/in/projecteka/consentmanager/consent/ConsentRequestNotificationListener.java  -2
+ src/main/java/in/projecteka/consentmanager/dataflow/DataFlowBroadcastListener.java  -2
+ src/main/java/in/projecteka/consentmanager/consent/HipConsentNotificationListener.java  -2
+ src/main/java/in/projecteka/consentmanager/consent/PostConsentRequest.java  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/ProjectEKA/consent-manager/pullRequest?prid=5556146&bid=18221425)

**Comment by ganesanarun** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity decreasing per file
==============================
+ src/main/java/in/projecteka/consentmanager/dataflow/DataFlowBroadcastListener.java  -2
+ src/main/java/in/projecteka/consentmanager/consent/ConsentRequestNotificationListener.java  -2
+ src/main/java/in/projecteka/consentmanager/consent/HipConsentNotificationListener.java  -2
+ src/main/java/in/projecteka/consentmanager/consent/PostConsentRequest.java  -2
+ src/main/java/in/projecteka/consentmanager/consent/ConsentNotificationPublisher.java  -2
+ src/main/java/in/projecteka/consentmanager/dataflow/PostDataFlowRequestApproval.java  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/ProjectEKA/consent-manager/pullRequest?prid=5556146&bid=18221425)

**Comment by ganesanarun** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity decreasing per file
==============================
+ src/main/java/in/projecteka/consentmanager/dataflow/DataFlowBroadcastListener.java  -2
+ src/main/java/in/projecteka/consentmanager/dataflow/PostDataFlowRequestApproval.java  -2
+ src/main/java/in/projecteka/consentmanager/consent/PostConsentRequest.java  -2
+ src/main/java/in/projecteka/consentmanager/consent/ConsentRequestNotificationListener.java  -2
+ src/main/java/in/projecteka/consentmanager/consent/HipConsentNotificationListener.java  -2
+ src/main/java/in/projecteka/consentmanager/consent/ConsentNotificationPublisher.java  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/ProjectEKA/consent-manager/pullRequest?prid=5556146&bid=18221425)

**Comment by ganesanarun** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity decreasing per file
==============================
+ src/main/java/in/projecteka/consentmanager/consent/ConsentNotificationPublisher.java  -2
+ src/main/java/in/projecteka/consentmanager/consent/PostConsentRequest.java  -2
+ src/main/java/in/projecteka/consentmanager/consent/HipConsentNotificationListener.java  -2
+ src/main/java/in/projecteka/consentmanager/dataflow/DataFlowBroadcastListener.java  -2
+ src/main/java/in/projecteka/consentmanager/consent/ConsentRequestNotificationListener.java  -2
+ src/main/java/in/projecteka/consentmanager/dataflow/PostDataFlowRequestApproval.java  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/ProjectEKA/consent-manager/pullRequest?prid=5556146&bid=18221425)

**Comment by ganesanarun** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity decreasing per file
==============================
+ src/main/java/in/projecteka/consentmanager/dataflow/DataFlowBroadcastListener.java  -2
+ src/main/java/in/projecteka/consentmanager/consent/ConsentRequestNotificationListener.java  -2
+ src/main/java/in/projecteka/consentmanager/consent/ConsentNotificationPublisher.java  -2
+ src/main/java/in/projecteka/consentmanager/consent/HipConsentNotificationListener.java  -2
+ src/main/java/in/projecteka/consentmanager/dataflow/PostDataFlowRequestApproval.java  -2
+ src/main/java/in/projecteka/consentmanager/consent/PostConsentRequest.java  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/ProjectEKA/consent-manager/pullRequest?prid=5556146&bid=18221425)

**Comment by ganesanarun** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity decreasing per file
==============================
+ src/main/java/in/projecteka/consentmanager/consent/ConsentNotificationPublisher.java  -2
+ src/main/java/in/projecteka/consentmanager/consent/ConsentRequestNotificationListener.java  -2
+ src/main/java/in/projecteka/consentmanager/consent/PostConsentRequest.java  -2
+ src/main/java/in/projecteka/consentmanager/dataflow/DataFlowBroadcastListener.java  -2
+ src/main/java/in/projecteka/consentmanager/dataflow/PostDataFlowRequestApproval.java  -2
+ src/main/java/in/projecteka/consentmanager/consent/HipConsentNotificationListener.java  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/ProjectEKA/consent-manager/pullRequest?prid=5556146&bid=18221425)

**Comment by ganesanarun** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity decreasing per file
==============================
+ src/main/java/in/projecteka/consentmanager/consent/ConsentRequestNotificationListener.java  -2
+ src/main/java/in/projecteka/consentmanager/consent/PostConsentRequest.java  -2
+ src/main/java/in/projecteka/consentmanager/consent/HipConsentNotificationListener.java  -2
+ src/main/java/in/projecteka/consentmanager/dataflow/DataFlowBroadcastListener.java  -2
+ src/main/java/in/projecteka/consentmanager/consent/ConsentNotificationPublisher.java  -2
+ src/main/java/in/projecteka/consentmanager/dataflow/PostDataFlowRequestApproval.java  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/ProjectEKA/consent-manager/pullRequest?prid=5556146&bid=18221425)

**Pull request merged**